docs/onboarding.md

```diff
@@ -0,0 +1,139 @@
+
+## Other Considersations
+
+### Why use custom frameworks?
+
+There are several reasons this approach was taken for implementing the Digital Marketplace. The web app was originally built during a five-month contract by a vendor, Real Folk, in 2019/20. There were no software engineers employed by the province at the time. The original vendor, Real Folk, realized the importance of building an application with a minimal failure rate (i.e. minimize the risk of run-time exceptions) because of (1) the province's lack of internal resources to maintain the application if something went wrong, and (2) the lengthy process involved with procuring a vendor to fix any issues that may arise.
+
+So, to avoid a period of prolonged downtime, the decision was made the implement the application with type safety as the most important architectural outcome. Doing so would almost completely eliminate the risk of run-time exceptions while significantly reducing the need for unit tests. At the time, the Node.js ecosystem did not have many strongly-typed application frameworks (modules were often "JavaScript-first" and `any` was pervasively used in type interface files), so the Real Folk development team built the back-end and front-end frameworks from scratch. This also offered the benefit of a "controlled dependency" that wouldn't be deprecated outside the control of the province, in alignment with the above-mentioned architectural goals.
+
+### Algebraic Data Types
+
+The application employs Algebraic Data Types (A.K.A ADTs) throughout the source code. ADTs are a fundamental building block of functional programming, inspiring their usage here. It is recommended that developers learn about ADTs and how they can be used in order to effectively contribute to this codebase.
+
+# Front-End Architecture
+
+The front-end framework's source code is located in the `src/front-end/typescript/lib/framework` directory. The framework is effectively a TypeScript port of the Elm architecture, mirroring the following concepts:
+
+- Components with `init`, `update` and `view` functions, and `State` (A.K.A `Model` in Elm) and `Msg` types.
+- Managed effects via `Cmd`.
+- Immutable representation of state.
+
+However, it differs from the Elm architecture in several ways, mainly due to the nature of TypeScript/JavaScript and the scope of the Digital Marketplace web application:
+
+- Several component "types": `App`, `Page`, `Global`, `Base`.
+  - `Page` components are coupled to the Digital Marketplace application's style guide and design system, implementing functionality like alerts, modals, sidebars, etc.
+- A new `Process` type that represents a running `App` component, allowing developers to introspect running applications in debug mode.
+- A prescriptive approach to component composition via helper functions exposed by the framework.
```

Comment:
It would be nice if we had a series of steps to determine where these go; feels like there are several helper functions that do the same thing